model/riscv_vmem_sv39.sail

```diff
@@ -1,5 +1,18 @@
 /* Sv39 address translation for RV64. */
 
+val napot_bits39 : (bits(44), nat) -> nat
+function napot_bits39 (v, n) = {
+  let mask = shiftl(v ^ v ^ EXTZ(0b1), n + 1) - 1;
+  let pattern = shiftl(v ^ v ^ EXTZ(0b1), n);
+  if n > 8 then {
+    0
+  } else if (v & mask) == pattern then {
+    n + 1
+  } else {
+    napot_bits39(v, n + 1)
```

Comment:
The recursion seems unnecessarily complex. What are you actually trying to compute? Would a loop suit you better or can you do away with even that?

Reply:
Please forgive my Sail naivety.  This is really just "find last zero" aka "count trailing zeros + 1".  If there's a simpler way to write that, great.

Reply:
There's a `count_leading_zeros` in the standard library (in fact it's a builtin) but seemingly no trailing ones. There's also the following commented-out version (for if it weren't a builtin):
```
function count_leading_zeros x = {
  foreach (i from ('N - 1) to 0 by 1 in dec)
    if [x[i]] == [bitone] then return 'N - i - 1;
  return 'N;
}
```
so I imagine something like the following would work:
```
val count_trailing_zeros = "count_trailing_zeros" : forall 'N , 'N >= 1. bits('N) -> {'n, 0 <= 'n <= 'N . atom('n)}
function count_trailing_zeros x = {
  foreach (i from 0 to ('N - 1) by 1)
    if [x[i]] == [bitone] then return i;
  return 'N;
}
```
Not sure what the square brackets around `x[i]` and `bitone` are for, mind, I'd expect it to work without.

Reply:
This is probably very naive, but can't you just perform count_trailing_zeroes(~x)